src/protocol.rs

```diff
@@ -18,16 +20,63 @@ use crate::util::Hash;
 pub(crate) const MAX_MESSAGE_SIZE: usize = 1024 * 1024 * 100;
 
 /// Protocol version
-pub const VERSION: u64 = 2;
+pub const VERSION: u64 = 3;
 
-#[derive(Deserialize, Serialize, Debug, PartialEq, Eq, Clone, MaxSize)]
+/// Connection tokens are 32 byte values used to authorize connections.
+//
+/// These are currently unused, but exist to distinguish connection authorization
+/// from request authorization.
+pub type ConnectionToken = [u8; 32];
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq, MaxSize, Clone)]
 pub(crate) struct Handshake {
     pub version: u64,
+    pub conn_token: Option<ConnectionToken>,
```

Comment:
Please leave this out for now. This will very likely be stored and validated at a different point.